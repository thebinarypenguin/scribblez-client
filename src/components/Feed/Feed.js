import React from 'react';
import UserContext from '../../contexts/UserContext';
import TopBar from '../TopBar/TopBar';
import FeedList from '../FeedList/FeedList';
import Utils from '../../services/Utils';

import './Feed.css';

class Feed extends React.Component {

  static contextType = UserContext;

  static contextType = UserContext;

  constructor(props) {
    super(props);

    this.generateLinks = this.generateLinks.bind(this);
  }

  generateLinks() {

    let links = [];

    if (Utils.isTokenActive(window.localStorage.getItem('token'))) {

      links = [
        { name: 'Feed',     target: '/feed'     },
        { name: 'Notes',    target: '/notes'    },
        { name: 'Sign Out', target: '/sign-out' },
      ];

    } else {

      links = [
        { name: 'Feed',    target: '/feed'    },
        { name: 'Sign Up', target: '/sign-up' },
        { name: 'Sign In', target: '/sign-in' },
      ];
    }

    return links;
  }

  render() {

    return (
      <>
        <TopBar links={this.generateLinks()} />
        <FeedList />
      </>
    );
  }
}

export default Feed;
