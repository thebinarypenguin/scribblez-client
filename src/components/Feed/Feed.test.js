import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from "react-router-dom";
import Feed from './Feed';

it('renders without crashing', () => {

  const container = document.createElement('div');

  const subject = (
    <MemoryRouter>
      <Feed />
    </MemoryRouter>
  )

  ReactDOM.render(subject, container);
  ReactDOM.unmountComponentAtNode(container);
});
