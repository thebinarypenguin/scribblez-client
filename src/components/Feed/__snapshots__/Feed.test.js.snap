// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  id="Feed"
>
  <div
    id="TopBar"
  >
    <div
      className="Container"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <h1>
          Scribblez
        </h1>
      </a>
      <ul>
        <li>
          <a
            href="/feed"
            onClick={[Function]}
          >
            Feed
          </a>
        </li>
        <li>
          <a
            href="/sign-up"
            onClick={[Function]}
          >
            Sign Up
          </a>
        </li>
        <li>
          <a
            href="/sign-in"
            onClick={[Function]}
          >
            Sign In
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="Container"
  >
    <ul
      id="FeedList"
    />
  </div>
</div>
`;
